dataprep/eda/distribution/render.py

```diff
@@ -276,7 +276,7 @@ def wordfreq_viz(
     nrows: int,
     plot_width: int,
     plot_height: int,
-    show_yticks: bool,
+    cfg: Config,
```

Comment:
Since it only need involves one or two parameters from `config` (`cfg.wordfreq.color` in this example), it would be more clear to pass the related parameters only instead of `config`. This also applies to other functions.

Reply:
I thought Jiannan has talked about supporting many Bokeh parameters in the future, which means we would need to access multiple attributes of `cfg` in each plotting function. So I thought we could set the precedent now of passing in only `cfg`. Does this change your mind @jinglinpeng ?

Reply:
Yes. Thanks for the information! How about only pass the related config, i.e., `cfg.wordfreq` in this case?

Reply:
Is there any reason you want to pass `cfg.wordfreq` rather than `cfg`? In both cases, they don't pass only the related parameters, which was the problem you initially identified from what I understand.

Reply:
This is because `wordfreq_viz` function only involves `wordfreq` related config, which is in `cfg.wordfreq`, while `cfg` contain all parameters from all visualizations.

Reply:
I know what params `wordfreq_viz` uses and that `cfg.wordfreq` holds these params while `cfg` holds all params, I'm trying to understand why this approach is better. Sorry I haven't really done any software dev, particularly on this scale.

However, in other places, we pass the full config even though only params from one or a subset of the attributes are used. E.g. for insight related functions such as [here](https://github.com/sfu-db/dataprep/blob/5878dc0fc9a24eff81ec09dd77bd56a3523d7e7b/dataprep/eda/distribution/compute/overview.py#L206), [here](https://github.com/sfu-db/dataprep/blob/5878dc0fc9a24eff81ec09dd77bd56a3523d7e7b/dataprep/eda/distribution/render.py#L1562), we only use `cfg.insight` but pass `cfg`. So I'm wondering if there is any inconsistency here with your proposed approach.

Another thing is that `cfg.wordfreq` is class `WordFrequency` not `Config`, so for type specification in render.py we would need to import `WordFrequency, Bar, Pie`, ....  And I'm wondering if we will ever need to access any global or insight parameters from `cfg` in the plotting functions. Not sure if any of the previous two points are relevant.

Reply:
I think it's common to only pass minimal parameters, e.g., https://github.com/apache/calcite/pull/2187#discussion_r502811366. It would be easier to debug as one will know that this function only related to `WordFrequency` and do not need to worry that other configs in `config` affects this function. 

For your example, it also accessed `cfg.stats.enable`, `cfg.bar.enable`,..., so I think passing `cfg` would be more convenient. 

I'm not sure whether we will access insight from `cfg` in the future. For now, it seems we only access global parameter `height` and `width` (but these two are passed explicitly and not read from `cfg`). If you're leaving space for the possible access of other parameters in `config`, I think it's OK to pass `cfg`.